src/ottplan.java

```diff
@@ -0,0 +1,141 @@
+import java.text.DecimalFormat;
+import java.util.*;
+abstract class Subscription{
+    protected String resolution;
+    protected String planName;
+    protected int noOfViewers;
+    protected double amount;
+    protected String dis;
+    abstract void calculate(boolean a);
+    void display()
+    {
+        DecimalFormat df=new DecimalFormat("##.00");
+        System.out.println(dis);
+        System.out.println("Plane Name:"+planName);
+        System.out.println("No. Of Concurrent Viewers:"+noOfViewers);
+        System.out.println("Supported Resolution:"+resolution);
+        System.out.println("Total Amount:"+df.format(amount));
+    }
+}
+class Prime extends Subscription{
+    @Override
+    public void calculate(boolean discount){
+        this.planName="Prime";
+        this.noOfViewers=3;
+        this.resolution="720p";
+        this.amount=999;
+        this.amount+=(0.0875*this.amount);
+        if(discount){
+            this.dis="Eligible for Discount";
+            this.amount-=100;
+        }
+        else
+            this.dis="Not Eligible for Discount";
+    }
+}
+class Premium extends Subscription{
+    @Override
+    public void calculate(boolean discount) {
+        this.planName="Premium";
+        this.noOfViewers=5;
+        this.resolution="1080p";
+        this.amount=1299;
+        this.amount+=(0.1425*this.amount);
+        if(discount){
+            this.dis="Eligible for Discount";
+            this.amount-=100;
+        }
+        else
+            this.dis="Not Eligible for Discount";
+    }
+}
+class Platinum extends Subscription{
+    @Override
+    public void calculate(boolean discount) {
+        this.planName="Platinum";
+        this.noOfViewers=10;
+        this.resolution="4k";
+        this.amount=1499;
+        this.amount+=(0.175*this.amount);
+        if(discount){
+            this.dis="Eligible for Discount";
+            this.amount-=100;
+        }
+        else
+            this.dis="Not Eligible for Discount";
+    }
+}
+class Factory{
+    static Map<String,Subscription> planMap=new HashMap<>();
+    static {
+        planMap.put("prime",new Prime());
+        planMap.put("premium",new Premium());
+        planMap.put("platinum",new Platinum());
+    }
+    public static Subscription getPlan(String operator)
+    {
+        return planMap.get(operator.toLowerCase());
+    }
+}
+public class ottplan {
+    boolean notConsecutive(String name)
```

Comment:
Functions that return boolean values should be prefixed with 'is' or equivalent verbs.

Like `hasConsecutiveSameCharacters(String name)`. The function name should tell you what the function would do.

Also avoid negatives while naming boolean functions. It'd become confusing with the output of the function is false. 

`isConsecutive` is better than `isNotConsecutive`